### ISSUE TITLE
Replace usage of COUNT(*) from SQLite storage

### ISSUE THREAD

**Issue by Onager** [log2timeline/plaso#1646]
Observed processing a Windows image on a fairly constrained machine. Symptoms:

- Processing not complete despite a long run time
- Most workers idle, some killed
- Large number of tasks pending merge
- Main thread not reporting merges of sources or events

**Comment by Onager** [comment]
Debugging with gdb, main process is in running in the sqlite storage, doing a "SELECT COUNT(*) FROM event_source" here https://github.com/log2timeline/plaso/blob/20171231/plaso/storage/sqlite_file.py#L221

After running this for a few more seconds, it looks like thread is still in the same place, which is a bit surprising: 
index = 656060
sequence_number = 65606

This seems to be a select on the session file, the main process doesn't seem to have task file open.

Backtrace:
```
    File "/usr/lib/python2.7/dist-packages/plaso/storage/sqlite_file.py", line 221, in _GetAttributeContainerByIndex
    self._cursor.execute(query)
  File "/usr/lib/python2.7/dist-packages/plaso/storage/sqlite_file.py", line 647, in GetEventSourceByIndex
    return self._GetAttributeContainerByIndex('event_source', index)
  File "/usr/lib/python2.7/dist-packages/plaso/storage/interface.py", line 1300, in GetNextWrittenEventSource
    self._written_event_source_index)
  File "/usr/lib/python2.7/dist-packages/plaso/multi_processing/task_engine.py", line 166, in _FillEventSourceHeap
    event_source = storage_writer.GetNextWrittenEventSource()
  File "/usr/lib/python2.7/dist-packages/plaso/multi_processing/task_engine.py", line 427, in _ScheduleTasks
    self._FillEventSourceHeap(storage_writer, event_source_heap)
  File "/usr/lib/python2.7/dist-packages/plaso/multi_processing/task_engine.py", line 323, in _ProcessSources
    self._ScheduleTasks(storage_writer)
  File "/usr/lib/python2.7/dist-packages/plaso/multi_processing/task_engine.py", line 827, in ProcessSources
    filter_find_specs=filter_find_specs)
  File "/usr/lib/python2.7/dist-packages/plaso/cli/log2timeline_tool.py", line 481, in ExtractEventsFromSources
    worker_memory_limit=self._worker_memory_limit)
  File "/usr/bin/log2timeline.py", line 54, in Main
    tool.ExtractEventsFromSources()
  File "/usr/bin/log2timeline.py", line 68, in <module>
    if not Main():
```

**Comment by Onager** [comment]
I suspect we'll have to refactor the way the storage works here, and potentially the logic around

**Comment by Onager** [comment]
Thinking more, select (*) is a bad plan: https://stackoverflow.com/questions/8988915/sqlite-count-slow-on-big-tables

**Comment by Onager** [comment]
This was fixed in #1647.
